### PR TITLE
feat(integrations): Gitlab on Prem Client

### PR DESCRIPTION
**Pull request by lauryndbrown**
Creates the client for gitlab on prem using much of the previous gitlab plugin's client. It does not currently do refresh auth. That can be covered by a pr to follow.

### CODE REVIEW

**Review by MeredithAnya — COMMENTED** [review]
I think we should change this to handle expired tokens by checking the response, and if it has expired, then make a new token and try the request again. This is cause Gitlab seems to only give us `created_at` and not `expires_in` (even though we could assume its `created_at` + `7200`)

cc @mnoble

File: src/sentry/integrations/gitlab/client.py [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+from six.moves.urllib.parse import quote
+
+from sentry.integrations.client import ApiClient, OAuth2RefreshMixin
+from sentry.integrations.exceptions import ApiError
+
+
+class GitLabApiClient(ApiClient, OAuth2RefreshMixin):
+    api_version = '/v4/'
+
+    def __init__(self, base_url, access_token):
```
Comment (MeredithAnya) [inline src/sentry/integrations/gitlab/client.py]: I think we might have to pass the integration here, or at least the `integration.metadata['verify_ssl']` for all the request. In GHE we have it hard coded to `False` but that will change too.

**Review by MeredithAnya — DISMISSED** [review]

File: src/sentry/integrations/gitlab/client.py (hunk 1 above)
Comment (MeredithAnya) [reply src/sentry/integrations/gitlab/client.py]: Hopefully I'll get the Install flow out soon (hopefully merged tomorrow) so we can test this

**Comment by lauryndbrown** [comment]
@MeredithAnya because all the other OAuth integrations use expires, and I feel like when the tokens expire is not something that's likely to change for gitlab, I think it's fine to just do created_at+time.

**Comment by MeredithAnya** [comment]
>  I think it's fine to just do created_at+time.

If we wanna do it this way, we should add that in https://github.com/getsentry/sentry/blob/master/src/sentry/integrations/gitlab/integration.py#L160

**Comment by mnoble** [comment]
Seems like token expiration values depend on how the user has configured GitLab:

https://stackoverflow.com/questions/39264452/gitlab-oauth-access-token-validity

Might be good to go the "Get a 401, refresh the token" route, since it seems like the expiration can vary.

**Comment by lauryndbrown** [comment]
@mnoble oh I didn't realize that :o ok sounds good then. @MeredithAnya I will have a different pr to manage refreshing the auth.

**Review by MeredithAnya — APPROVED** [review]

**Pull request merged**